src/main.js

```diff
@@ -0,0 +1,54 @@
+//base class
+class Vehicle{
+    constructor(color,type,year){
+        this.name='Vehicle';
+        this.color =color;
+        this.type =type;
+        this.year =year;
+
```

Comment:
These variables are not private

Reply:
Also, with the way you assigned 'Vehicle' to this.name. that means all classes inherited will always have 'vehicle' as the name. I'm not sure that's what you intended to do.